work.py

```diff
@@ -417,7 +424,8 @@ def sync_timesheetline(cls, activities):
                 if getattr(timesheet_line, attribute, None) != value:
                     setattr(timesheet_line, attribute, value)
 
-            if timesheet_line.work != activity.resource.timesheet_works[0]:
+            if (timesheet_line.work and activity.resource.timesheet_works
+                    and (timesheet_line.work != activity.resource.timesheet_works[0])):
```

Comment:
The problem with handling this here is that we don't warn the user and we don't change the work, which is wrong. I'll propose another patch.